File: 268-missing-number/missing-number.cpp
class Solution {
public:
    int missingNumber(vector<int>& nums) {
        int n=0;
        int p=0;
        for(int i=0;i<=nums.size();i++){
            n=n^i;
            if(i<nums.size()){
                 p=nums[i]^p;
            }
           
        }
        return p^n;
    
    }
};

// class Solution {
// public:
//     int missingNumber(vector<int>& nums) {
//         set<int>set(nums.begin(),nums.end());
//         int n=nums.size();
//         for(int i=0;i<=n;i++){
//             if(set.find(i)==set.end()){
//                 return i;
//             }
//         }
//         return -1;
//     }
// };

File: 268-missing-number/missing-number.java
class Solution {
    public int missingNumber(int[] nums) {
        int n=0;
        int p=0;
       int m=nums.length;
       for(int i=0;i<m;i++){
         n=n^i;
            p=p^nums[i];
    }
    n=n^m;
    return p^n;
    }
}